tests/bdd/features/volume/resize/resize_offline.feature

```diff
@@ -0,0 +1,46 @@
+Feature: Volume resize
+
+    Background:
+        Given a deployer cluster
+
+    Scenario Outline: expand an unpublished volume
+        Given an unpublished volume with <repl_count> replicas and all are healthy
+        When the volume is expanded
+        Then all the replicas of volume should be resized to new capacity
+        And the volume expand status should be success
+        And the volume is expanded to the new capacity
+        Examples:
+        | repl_count |
+        | 1          |
+        | 3          |
+
+
+    Scenario: expand a published volume
+        Given a published volume with more than one replica and all are healthy
+        When the volume is expanded
+        Then the volume expand status should be failure
+        And the failure reason should be volume-in-use precondition
+
+
+    Scenario: expand an unpublished volume and then make it published
+        Given an unpublished volume with more than one replica and all are healthy
+        When the volume is expanded
+        Then all the replicas of volume should be resized to new capacity
+        When the volume is published
+        Then the volume should get published with expanded capacity
+        And the new capacity should be available for the application to use
+
+
+    Scenario: expand an unpublished volume having a not-online replica
+        Given an unpublished volume with more than one replica
+        And one of the replica is not in online state
+        When the volume is expanded
+        Then the volume expand status should be failure
```

Comment:
Let's also ensure that when the replica becomes online it's rebuilt
Also would be good to check also what happens if we increase the replica count

Reply:
But for unpublished volume rebuild won't be there after replica becomes online.

Reply:
Yes we have to add a clause where the volume becomes republished.

Reply:
Though it overlaps a bit with other existing scenarios, I'll add this here
```
        When the replica comes online again
        And we issue a volume expand request
        Then the request should succeed
       And all the replicas of the volume should be resized to the new capacity
        When the volume is published
        Then the onlined replica should be rebuilt
```

Reply:
hmm, this doesn't seem to add up here cleanly @tiagolobocastro . The publish here has to happen after replica comes online because resize will fail for published volume, this means rebuild won't happen. If thinking about republish, I'll add a new scenario instead of modifying this one. The new one will be.

```
    Scenario: Expand a published volume after unpublishing it while having an offline replica
        Given a published volume with more than one replica and all are healthy
        And the volume is receiving IO
        And one of the replica goes offline
        And the volume is unpublished
        When the replica comes online again
        And we issue a volume expand request
        Then the request should succeed
        And all the replicas of the volume should be resized to the new capacity
        When the volume is republished
        Then the volume should get published with expanded capacity
        And the onlined replica should be rebuilt
```

Reply:
I'll add this in next set of test changes, it's taking some trouble to work seamlessly. This PR can go ahead meanwhile.

Reply:
ok, worked now. Added a new test scenario.